addons/mass_mailing/tests/test_mailing_internals.py

```diff
@@ -313,6 +313,62 @@ def test_mailing_computed_fields_default(self):
         })
         self.assertEqual(literal_eval(mailing.mailing_domain), [('email', 'ilike', 'test.example.com')])
 
+    @users('user_marketing')
+    def test_mailing_computed_fields_default_email_from(self):
+        # Testing if the email_from is correctly computed when an
+        # alias domain for the company is set
+
+        # Case where alias domain is set and no outgoing email server for mass_mailing
+        mailing = self.env['mailing.mailing'].create({
+            'name': 'TestMailing no default mail server for mass mailing',
+            'subject': 'Test',
+        })
+        self.assertEqual(mailing.email_from, self.env.user.email_formatted)
+
+        # Case where alias domain is set and there is a default outgoing email server
+        # for mass mailing. from_filter matches domain of company alias domain
+        # before record creation
+        mass_mailing_mail_server = self.env['ir.mail_server'].sudo().create({
+            'name' : 'mass_mailing_test',
+            'from_filter' : 'test.mycompany.com',
+            'smtp_host' : 'not_real@smtp.com'
+        })
+
+        self.env['res.config.settings'].sudo().create({
+            'mass_mailing_mail_server_id' : mass_mailing_mail_server.id,
+            'mass_mailing_outgoing_mail_server' : True
+        }).execute()
+
+        mailing_alias_domain = self.env['mailing.mailing'].create({
+            'name': 'TestMailingAliasDomain matches',
+            'subject': 'Test',
+        })
+        self.assertEqual(mailing_alias_domain.email_from, self.env.user.company_id.alias_domain_id.default_from_email)
```

Comment:
While we are here, could test 'mail_server_id' field as it is quite part of the test

Reply:
Thx for the review as usual. Added an assert for the mail_server_id (hoping that I understood your comment)

Reply:
Indeed :)